Move current damage logic out of the talent loop

Jasperrr91 opened the pull request:
Move logic to get the current damage levels out of the loop that checks each talent, because this value will be the same for each talent. This should save some processing power.

Jasperrr91 committed:
Move current damage logic out of the talent loop

Move logic to get the current damage levels out of the loop that checks each talent, because this value will be the same for each talent. This should save some processing power.

gragatrim reviewed (changes requested):
This won't work as $talent isn't defined yet for $current_talent_damage. Move $current_talent_damage to be under the if debug check, it's only used/needed there anyway. The $current_total_damage I think should work being moved out, you are right, it should be exactly the same every time.

Jasperrr91 committed:
Move current_talent_damage to relevant logic

Move the current_talent_damage to under the debug statement, the only place where it's used. Thus only calculating this value if actually needed.

Jasperrr91 commented:
> This won't work as $talent isn't defined yet for $current_talent_damage. Move $current_talent_damage to be under the if debug check, it's only used/needed there anyway. The $current_total_damage I think should work being moved out, you are right, it should be exactly the same every time.

Good call, completely overlooked that it was in fact dependent on the talent. Updated the code!

gragatrim reviewed (approved):
Looks great, thanks!
